spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/inbound/ReactiveMongoDbMessageSource.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.inbound;
+
+import org.reactivestreams.Publisher;
+
+import org.springframework.data.mongodb.ReactiveMongoDatabaseFactory;
+import org.springframework.data.mongodb.core.ReactiveMongoOperations;
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import org.springframework.data.mongodb.core.query.BasicQuery;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.expression.Expression;
+import org.springframework.expression.TypeLocator;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.expression.spel.support.StandardTypeLocator;
+import org.springframework.integration.endpoint.AbstractMessageSource;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.mongodb.support.MongoHeaders;
+import org.springframework.integration.support.AbstractIntegrationMessageBuilder;
+import org.springframework.util.Assert;
+
+import com.mongodb.DBObject;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * @author David Turanski
+ *
+ * @since 5.3
+ */
+public class ReactiveMongoDbMessageSource extends AbstractMessageSource<Publisher<?>> {
+	private final Expression queryExpression;
+
+	private volatile Expression collectionNameExpression = new LiteralExpression("data");
+
+	private volatile StandardEvaluationContext evaluationContext;
+
+	private volatile ReactiveMongoOperations reactiveMongoTemplate;
+
+	private volatile MongoConverter mongoConverter;
+
+	private volatile ReactiveMongoDatabaseFactory reactiveMongoDatabaseFactory;
+
+	private volatile boolean initialized = false;
+
+	private volatile Class<?> entityClass = DBObject.class;
+
+	private volatile boolean expectSingleResult = false;
+
+	/**
+	 * Creates an instance with the provided {@link ReactiveMongoDatabaseFactory} and SpEL expression
```

Comment:
Also copied from the non reactive version

Reply:
David, we try to fix smells in the old classes when we touch them, but when we start new classes it would be better to have them without flaws already.

I have something about Code Style in here: https://github.com/spring-projects/spring-integration/wiki/Spring-Integration-Framework-Code-Style

Let me know if I am too nit-pick with you and that one has to be update somehow to let us to avoid such a conversion in the future.

Thanks

Reply:
I agree, this was a copy and paste job, so I didn't pay too much attention to those details